orchagent/nhgorch.cpp

```diff
@@ -0,0 +1,1386 @@
+#include "nhgorch.h"
+#include "switchorch.h"
+#include "neighorch.h"
+#include "crmorch.h"
+#include "routeorch.h"
+#include "bulker.h"
+#include "logger.h"
+#include "swssnet.h"
+
+extern sai_object_id_t gSwitchId;
+
+extern PortsOrch *gPortsOrch;
+extern CrmOrch *gCrmOrch;
+extern NeighOrch *gNeighOrch;
+extern SwitchOrch *gSwitchOrch;
+extern RouteOrch *gRouteOrch;
+
+extern size_t gMaxBulkSize;
+
+extern sai_next_hop_group_api_t* sai_next_hop_group_api;
+extern sai_next_hop_api_t*         sai_next_hop_api;
+extern sai_switch_api_t*            sai_switch_api;
+
+unsigned int NextHopGroup::m_count = 0;
+
+/* Default maximum number of next hop groups */
+#define DEFAULT_NUMBER_OF_ECMP_GROUPS   128
+#define DEFAULT_MAX_ECMP_GROUP_SIZE     32
+
+NhgOrch::NhgOrch(DBConnector *db, string tableName) :
+    Orch(db, tableName)
+{
+    SWSS_LOG_ENTER();
+
+    /* Get the switch's maximum next hop group capacity. */
+    sai_attribute_t attr;
+    attr.id = SAI_SWITCH_ATTR_NUMBER_OF_ECMP_GROUPS;
+
+    sai_status_t status = sai_switch_api->get_switch_attribute(gSwitchId,
+                                                               1,
+                                                               &attr);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Failed to get switch attribute number of ECMP groups. \
+                       Use default value. rv:%d", status);
+        m_maxNhgCount = DEFAULT_NUMBER_OF_ECMP_GROUPS;
+    }
+    else
+    {
+        m_maxNhgCount = attr.value.s32;
+
+        /*
+         * ASIC specific workaround to re-calculate maximum ECMP groups
+         * according to diferent ECMP mode used.
+         *
+         * On Mellanox platform, the maximum ECMP groups returned is the value
+         * under the condition that the ECMP group size is 1. Deviding this
+         * number by DEFAULT_MAX_ECMP_GROUP_SIZE gets the maximum number of
+         * ECMP groups when the maximum ECMP group size is 32.
+         */
+        char *platform = getenv("platform");
+        if (platform && strstr(platform, MLNX_PLATFORM_SUBSTRING))
+        {
+            SWSS_LOG_NOTICE("Mellanox platform - divide capacity by %d",
```

Comment:
Please don't add extra logs or logic when you copy from existing code. If you are moving code, please move as-is

Reply:
Fixed (think it may have been removed from the original code at some point)